HSEARCH-5426 Adjust the count aggregation DSL (#4719)
hibernate/hibernate-search: documentation/src/main/asciidoc/public/reference/_search-dsl-aggregation.adoc

include::../components/_incubating-warning.adoc[]

Hibernate Search provides a set of most common metric aggregations such as `sum`, `min`, `max`, `count` and `avg`.

> These aggregations can be requested for fields of numerical or temporal types.

Reviewer: ```suggestion
These aggregations can be requested for fields of numerical or temporal types,
or, in the case of `count`, at the document level (counting documents).
```

----
include::{sourcedir}/org/hibernate/search/documentation/search/aggregation/AggregationDslIT.java[tags=count-documents]
----
<1> Start building the counts aggregation by calling `.count()`. This aggregation always return a `Long.class` value.

> <2> Request the documents count.

Reviewer: ```suggestion
<2> Request the document count.
```

I think grammatically this is never plural? Like "a three-child family"?

----
include::{sourcedir}/org/hibernate/search/documentation/search/aggregation/AggregationDslIT.java[tags=count]
----
<1> Start building the counts aggregation by calling `.count()`. This aggregation always return a `Long.class` value.

> <2> Request the count values aggregation by specifying the target field path to which you want to apply the aggregation.

Reviewer: ```suggestion
<2> Specify the path of the field whose values to count.
```

... Or something like that?

====

Additionally, it is possible to count only the distinct field values.

.Count the number of all different price value among all the science fiction books
====
[source, JAVA, indent=0, subs="+callouts"]
----
include::{sourcedir}/org/hibernate/search/documentation/search/aggregation/AggregationDslIT.java[tags=count-distinct]
----
<1> Start building the counts aggregation by calling `.count()`. This aggregation always return a `Long.class` value.

> <2> Request the count values aggregation by specifying the target field path to which you want to apply the aggregation.
> <3> Make the count values aggregations consider only distinct values.

Reviewer: ```suggestion
<2> Specify the path of the field whose values to count.
<3> Request a count of distinct values (ignoring duplicates).
```
